include/soci/type-conversion-traits.h

```diff
@@ -36,6 +36,29 @@ struct type_conversion
     }
 };
 
+// Most majority of database engines ignore unsigned integer related types.
+// This hack allows to trait these types as signed integers in the database.
+template<>
+struct type_conversion<unsigned long long>
```

Comment:
More importantly, the comment above speaks about "types", but the specialization here is just for a single type. I'm not sure if the comment should be trusted and we should also have specializations for `unsigned short`, `unsigned [int]` and `unsigned long` too or if the comment should be updated?